src/app/rpmostree-builtin-shlib-backend.c

```diff
@@ -32,21 +32,20 @@
 #include "rpmostree-builtins.h"
 #include "rpmostree-libbuiltin.h"
 #include "rpmostree-rust.h"
+#include "rpmostree-rpm-util.h"
 #include "rpmostree.h"
 #include "rpmostree-core.h"
 #include "src/lib/rpmostree-shlib-ipc-private.h"
 
 #include <libglnx.h>
 
 static gboolean
-send_memfd_result (int ret_memfd, GError **error)
+send_memfd_result (GSocket *ipc_sock, int ret_memfd, GError **error)
 {
   int fdarray[] = {ret_memfd, -1 };
   g_autoptr(GUnixFDList) list = g_unix_fd_list_new_from_array (fdarray, 1);
   g_autoptr(GUnixFDMessage) message = G_UNIX_FD_MESSAGE (g_unix_fd_message_new_with_fd_list (list));
-  g_autoptr(GSocket) ipc_sock = g_socket_new_from_fd (RPMOSTREE_SHLIB_IPC_FD, error);
-  if (!ipc_sock)
-    return FALSE;
```

Comment:
Yeah for sure but personally I find it's a *lot* of error-prone work to correctly apply fixups to prior commits in places where they might create a "tower of conflicts".  https://github.com/tummychow/git-absorb is useful but it's not an AI.  For sure around "absolutely critical code" I think it can be worth it (often seen in the Linux kernel) but I'm not sure we're at that level of reward/benefit for this code.  The chance someone is going to bisect is close to zero versus just reading our commits I think most bugs would jump out.